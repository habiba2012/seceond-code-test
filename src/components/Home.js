import React,{useState} from 'react';
import Tab from  './Tab';



const  Home= () => {
    const [open, setOpen] = useState(false)
 
    return (
        <div className="wrapper">
            <button className="btn-filter" 
            onClick={()=> setOpen(!open)}>
            <img src="assets/plus.svg" alt=""/> Add filter </button>
            {
                open &&
                <Tab />
            }
        </div>
    );
};

export default Home;