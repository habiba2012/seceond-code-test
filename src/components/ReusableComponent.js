import React from 'react';
import '../App.css'


const ReusableComponent = (
   {name, 
   image,
   options,
   logo, 
   handleOption, 
   handleIntegration,
   selectedList 
  
   }) => {

   
    return (
        <div>
          {options && logo?
          <div className="user-option">
             <img src={logo} alt="" className="user-integration"/> 
             <li onClick={handleIntegration}>{options}</li>
          </div>
       
        :
        <div className="user-option">
           <img src={image} className="user-profile" alt=""/>
           <li onClick={handleOption}>{name}</li>
           
        </div>
           }
         </div>
    );
};

export default ReusableComponent;