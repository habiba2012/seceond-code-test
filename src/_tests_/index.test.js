import React from 'react';
import { render } from '@testing-library/react';
import Home from '../components/Home';
import Tabs from '../components/Tab';
import ReusableComponent from '../components/ReusableComponent';

test('renders without throwing errors', () => {
    const wrapper = render(<Home />);
    expect(wrapper).toMatchSnapshot();
})

test('renders <Tab>', () => {
    const wrapper = render(<Tabs />);
    expect(wrapper).toMatchSnapshot();
})

test('renders <ReusableComponent>', () => {
    const wrapper = render(<ReusableComponent />);
    expect(wrapper).toMatchSnapshot();
})