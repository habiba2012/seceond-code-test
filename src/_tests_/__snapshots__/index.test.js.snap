// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <ReusableComponent> 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="user-option"
        >
          <img
            alt=""
            class="user-profile"
          />
          <li />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="user-option"
      >
        <img
          alt=""
          class="user-profile"
        />
        <li />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders <Tab> 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="bloc-tabs"
        >
          <button
            class="tabs active-tabs"
          >
            Users
          </button>
          <button
            class="tabs"
          >
            Integrations
          </button>
        </div>
        <div>
          <div
            class="search-container"
            style="display: flex;"
          >
            <img
              alt="search"
              class="searchIcon"
              src="/assets/search.svg"
            />
            <input
              class="searchBox"
              placeholder="Search option"
              type="text"
            />
          </div>
          <div>
            <div
              class="content-tabs"
            >
              <div
                class="content  active-content"
              >
                <div>
                  <div
                    class="user-option"
                  >
                    <img
                      alt=""
                      class="user-profile"
                      src="../assets/profile.svg"
                    />
                    <li>
                      Louie Popp
                    </li>
                  </div>
                </div>
              </div>
              <div
                class="content"
              >
                <div>
                  <div
                    class="user-option"
                  >
                    <img
                      alt=""
                      class="user-integration"
                      src="../assets/aCampaign.svg"
                    />
                    <li>
                      ActiveCampaign
                    </li>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div>
            <div
              class="content-tabs"
            >
              <div
                class="content  active-content"
              >
                <div>
                  <div
                    class="user-option"
                  >
                    <img
                      alt=""
                      class="user-profile"
                      src="../assets/profile.svg"
                    />
                    <li>
                      Jonas Rafn
                    </li>
                  </div>
                </div>
              </div>
              <div
                class="content"
              >
                <div>
                  <div
                    class="user-option"
                  >
                    <img
                      alt=""
                      class="user-integration"
                      src="../assets/googleAn.svg"
                    />
                    <li>
                      Google Analytics
                    </li>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div>
            <div
              class="content-tabs"
            >
              <div
                class="content  active-content"
              >
                <div>
                  <div
                    class="user-option"
                  >
                    <img
                      alt=""
                      class="user-profile"
                      src="../assets/profile.svg"
                    />
                    <li>
                      Fiona Rakipi
                    </li>
                  </div>
                </div>
              </div>
              <div
                class="content"
              >
                <div>
                  <div
                    class="user-option"
                  >
                    <img
                      alt=""
                      class="user-integration"
                      src="../assets/insta.svg"
                    />
                    <li>
                      Instagram
                    </li>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div>
            <div
              class="content-tabs"
            >
              <div
                class="content  active-content"
              >
                <div>
                  <div
                    class="user-option"
                  >
                    <img
                      alt=""
                      class="user-profile"
                      src="../assets/profile.svg"
                    />
                    <li>
                      Marti Navne
                    </li>
                  </div>
                </div>
              </div>
              <div
                class="content"
              >
                <div>
                  <div
                    class="user-option"
                  >
                    <img
                      alt=""
                      class="user-integration"
                      src="../assets/wocommerce.svg"
                    />
                    <li>
                      Woocommerce
                    </li>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div>
            <div
              class="content-tabs"
            >
              <div
                class="content  active-content"
              >
                <div>
                  <div
                    class="user-option"
                  >
                    <img
                      alt=""
                      class="user-profile"
                      src="../assets/profile.svg"
                    />
                    <li>
                      Kristoffer Degn
                    </li>
                  </div>
                </div>
              </div>
              <div
                class="content"
              >
                <div>
                  <div
                    class="user-option"
                  >
                    <img
                      alt=""
                      class="user-integration"
                      src="../assets/sopify.svg"
                    />
                    <li>
                      Sopify
                    </li>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="btn-remove"
        >
          
          
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="bloc-tabs"
      >
        <button
          class="tabs active-tabs"
        >
          Users
        </button>
        <button
          class="tabs"
        >
          Integrations
        </button>
      </div>
      <div>
        <div
          class="search-container"
          style="display: flex;"
        >
          <img
            alt="search"
            class="searchIcon"
            src="/assets/search.svg"
          />
          <input
            class="searchBox"
            placeholder="Search option"
            type="text"
          />
        </div>
        <div>
          <div
            class="content-tabs"
          >
            <div
              class="content  active-content"
            >
              <div>
                <div
                  class="user-option"
                >
                  <img
                    alt=""
                    class="user-profile"
                    src="../assets/profile.svg"
                  />
                  <li>
                    Louie Popp
                  </li>
                </div>
              </div>
            </div>
            <div
              class="content"
            >
              <div>
                <div
                  class="user-option"
                >
                  <img
                    alt=""
                    class="user-integration"
                    src="../assets/aCampaign.svg"
                  />
                  <li>
                    ActiveCampaign
                  </li>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="content-tabs"
          >
            <div
              class="content  active-content"
            >
              <div>
                <div
                  class="user-option"
                >
                  <img
                    alt=""
                    class="user-profile"
                    src="../assets/profile.svg"
                  />
                  <li>
                    Jonas Rafn
                  </li>
                </div>
              </div>
            </div>
            <div
              class="content"
            >
              <div>
                <div
                  class="user-option"
                >
                  <img
                    alt=""
                    class="user-integration"
                    src="../assets/googleAn.svg"
                  />
                  <li>
                    Google Analytics
                  </li>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="content-tabs"
          >
            <div
              class="content  active-content"
            >
              <div>
                <div
                  class="user-option"
                >
                  <img
                    alt=""
                    class="user-profile"
                    src="../assets/profile.svg"
                  />
                  <li>
                    Fiona Rakipi
                  </li>
                </div>
              </div>
            </div>
            <div
              class="content"
            >
              <div>
                <div
                  class="user-option"
                >
                  <img
                    alt=""
                    class="user-integration"
                    src="../assets/insta.svg"
                  />
                  <li>
                    Instagram
                  </li>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="content-tabs"
          >
            <div
              class="content  active-content"
            >
              <div>
                <div
                  class="user-option"
                >
                  <img
                    alt=""
                    class="user-profile"
                    src="../assets/profile.svg"
                  />
                  <li>
                    Marti Navne
                  </li>
                </div>
              </div>
            </div>
            <div
              class="content"
            >
              <div>
                <div
                  class="user-option"
                >
                  <img
                    alt=""
                    class="user-integration"
                    src="../assets/wocommerce.svg"
                  />
                  <li>
                    Woocommerce
                  </li>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="content-tabs"
          >
            <div
              class="content  active-content"
            >
              <div>
                <div
                  class="user-option"
                >
                  <img
                    alt=""
                    class="user-profile"
                    src="../assets/profile.svg"
                  />
                  <li>
                    Kristoffer Degn
                  </li>
                </div>
              </div>
            </div>
            <div
              class="content"
            >
              <div>
                <div
                  class="user-option"
                >
                  <img
                    alt=""
                    class="user-integration"
                    src="../assets/sopify.svg"
                  />
                  <li>
                    Sopify
                  </li>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="btn-remove"
      >
        
        
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders without throwing errors 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="wrapper"
      >
        <button
          class="btn-filter"
        >
          <img
            alt=""
            src="assets/plus.svg"
          />
           Add filter 
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="wrapper"
    >
      <button
        class="btn-filter"
      >
        <img
          alt=""
          src="assets/plus.svg"
        />
         Add filter 
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
